src/components/input/IconButton.tsx

```diff
@@ -43,37 +47,46 @@ const IconButton = function IconButton({
   size = 'md',
   title,
   variant = 'secondary',
+  unstyled = false,
 
   ...htmlAttributes
 }: IconButtonProps) {
+  const styled = !unstyled;
+  const themed = styled && variant !== 'custom';
+  const sized = styled && size !== 'custom';
+
   return (
-    <ButtonBase
+    <Button
       data-component="IconButton"
       {...htmlAttributes}
       classes={classnames(
-        'focus-visible-ring transition-colors whitespace-nowrap flex items-center',
-        'justify-center gap-x-2 rounded-sm',
-        {
+        styled && {
+          'focus-visible-ring transition-colors rounded-sm whitespace-nowrap':
+            true,
+          'flex items-center justify-center': true,
+          // Adapt styling when in an InputGroup
+        },
```

Comment:
I think the approach of `<condition> && {...}` makes sense when every entry in the object has their own different conditions, but they all share a common one.

In any other case, perhaps it's more clear to go with this:

```suggestion
        {
          'focus-visible-ring transition-colors rounded-sm whitespace-nowrap':
            styled,
          'flex items-center justify-center': styled,
          // Adapt styling when in an InputGroup
        },
```